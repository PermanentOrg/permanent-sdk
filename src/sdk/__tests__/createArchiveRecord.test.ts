import path from "node:path";
import fs from "node:fs";
import nock from "nock";
import { createArchiveRecord } from "..";

describe("createArchiveRecord", () => {
	it("should create an ArchiveRecord", async () => {
		const filePath = path.join(
			__dirname,
			"fixtures",
			"createArchiveRecord",
			"myFile.txt",
		);
		const { size: fileSize } = fs.statSync(filePath);

		nock("https://permanent.local")
			.post("/api/record/registerRecord", {
				s3url:
					"https://permanet.local/_slifty/unprocessed/12345678-e6ef-4238-85d3-20001abebb63",
				displayName: "myFile",
				parentFolderId: 1,
				uploadFileName: "myFile.txt",
				fileType: "text/plain",
				size: fileSize,
				failOnDuplicateName: false,
			})
			.replyWithFile(
				200,
				path.join(
					__dirname,
					"fixtures",
					"createArchiveRecord",
					"recordVo.json",
				),
				{
					"Content-Type": "application/json",
				},
			);

		const archiveRecord = await createArchiveRecord(
			{
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			},
			{
				s3Url:
					"https://permanet.local/_slifty/unprocessed/12345678-e6ef-4238-85d3-20001abebb63",
				file: {
					size: fileSize,
					contentType: "text/plain",
				},
				item: {
					displayName: "myFile",
					fileSystemCompatibleName: "myFile.txt",
				},
				parentFolder: {
					id: 1,
				},
			},
		);

		expect(archiveRecord).toMatchSnapshot();
	});

	it("should pass the failOnDuplicateName parameter to the API", async () => {
		const filePath = path.join(
			__dirname,
			"fixtures",
			"createArchiveRecord",
			"myFile.txt",
		);
		const { size: fileSize } = fs.statSync(filePath);

		nock("https://permanent.local")
			.post("/api/record/registerRecord", {
				s3url:
					"https://permanet.local/_slifty/unprocessed/12345678-e6ef-4238-85d3-20001abebb63",
				displayName: "myFile",
				parentFolderId: 1,
				uploadFileName: "myFile.txt",
				fileType: "text/plain",
				size: fileSize,
				failOnDuplicateName: true,
			})
			.replyWithFile(
				200,
				path.join(
					__dirname,
					"fixtures",
					"createArchiveRecord",
					"recordVo.json",
				),
				{
					"Content-Type": "application/json",
				},
			);

		const archiveRecord = await createArchiveRecord(
			{
				bearerToken: "12345",
				baseUrl: "https://permanent.local/api",
			},
			{
				s3Url:
					"https://permanet.local/_slifty/unprocessed/12345678-e6ef-4238-85d3-20001abebb63",
				file: {
					size: fileSize,
					contentType: "text/plain",
				},
				item: {
					displayName: "myFile",
					fileSystemCompatibleName: "myFile.txt",
				},
				parentFolder: {
					id: 1,
				},
				failOnDuplicateName: true,
			},
		);

		expect(archiveRecord).toMatchSnapshot();
	});
});
