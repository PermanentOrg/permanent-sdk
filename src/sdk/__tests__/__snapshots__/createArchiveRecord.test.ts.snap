// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createArchiveRecord should create an ArchiveRecord 1`] = `
{
  "createdAt": 2023-01-23T21:02:40.000Z,
  "displayDate": 2023-01-23T21:02:40.000Z,
  "displayName": "test",
  "fileSystemCompatibleName": "test (1).txt",
  "fileSystemId": 14,
  "files": [],
  "id": 5,
  "status": "status.generic.undefined",
  "type": "type.record.document",
  "updatedAt": 2023-01-23T21:02:40.000Z,
}
`;
